aten/src/ATen/TensorUtils.cpp

```diff
@@ -151,6 +151,18 @@ void checkAllSameGPU(CheckedFrom c, ArrayRef<TensorArg> tensors) {
   checkAllSame(c, tensors, checkSameGPU);
 }
 
+void checkTensorsSameDevice(CheckedFrom c, const TensorArg& t1, const TensorArg& t2) {
+  TORCH_CHECK(
+    t1->get_device() == t2->get_device(),
+    "Expected tensor for ", t1, " to have the same device as tensor for ", t2,
+    "; but device ", t1->get_device(), " does not equal ", t2->get_device(),
+    " (while checking arguments for ", c, ")");
+}
+
+void checkAllSameDevice(CheckedFrom c, ArrayRef<TensorArg> tensors) {
```

Comment:
where will it be used?

Reply:
Thanks for taking a look, @anjali411!

It can be used in #56334, as well as in some functions in `LinearAlgebra.cpp`, `Linear.cpp`, et al.

For example, 
https://github.com/pytorch/pytorch/blob/0cc42809ce027def1ff8a95ad2c58b29251f7d42/aten/src/ATen/native/Linear.cpp#L649-L655
Here, `input2` might be on a different device in the above snippet, in which case, the above check wouldn't detect it.
So, `input2`'s device should also be compared.